docs/building-apps/frontends/custom-domains/using-custom-domains.mdx

```diff
@@ -0,0 +1,284 @@
+---
+keywords: [intermediate, tutorial, custom domains, using custom domains, configuring a domain for a canister, using your own domain, use domain]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Using custom domains
+
+<MarkdownChipRow labels={["Intermediate", "Tutorial"]} />
+
+By default, all canisters on the Internet Computer are accessible through the domain `icp0.io` and their canister ID. In addition to that default domain, one can also make a canister accessible under a custom domain using one of the following approaches:
+
+- [Register the domain with the boundary nodes](#custom-domains-on-the-boundary-nodes).
+- [Hosting the domain on your own infrastructure](#custom-domains-using-your-own-infrastructure).
+
+For both approaches, you need to acquire a domain through any registry, such as NameCheap or GoDaddy.
+
+The two approaches differ in the ease of use and configurability. When registering the domain with the boundary nodes, you simply have to configure the DNS records of the custom domain, and the boundary nodes take care of obtaining a certificate, renewing the certificate before its expiration, and SEO.
+
+When hosting the domain on your own infrastructure, you need to obtain and renew the certificates and provide your own infrastructure with support for the [HTTP gateway protocol](../../../references/http-gateway-protocol-spec.md). However, there is more configuration flexibility using this approach.
+
+## Register the domain with the boundary nodes
+
+By following the steps below, you can host your canister under your custom domain using the boundary nodes. First, the configuration steps are described, then an [example is used to illustrate these steps](#concrete-example), followed by some instructions on [troubleshooting](#troubleshooting).
+
+- #### Step 1: Configure the DNS record of your domain, which is denoted with `CUSTOM_DOMAIN`.
```

Comment:
Somehow the indentation is broken: compare with the original [here](https://internetcomputer.org/docs/current/developer-docs/web-apps/custom-domains/using-custom-domains). I like how you broke the "steps" within the steps down into its own bullets, but I would still indent it as before for better structure.

Reply:
Sorry, I just read your full instructions. We can fix that in a follow-up PR.